### DISCUSSION TITLE
Weird behaviour on contact - bodies diffusing into each other

### DISCUSSION THREAD

**Discussion by achuwilson** [bulletphysics/bullet3#3781] [General]
Hi, I am having a weird observation. I designed  a simple 2 finger gripper (URDF model with STL meshes) and attached it to the simulated Kuka robot and is trying to pick the checkered sphere given with pybullet_data. I close the gripper by position control, by commanding the fingers to move to the centre position.

When the fingers of the gripper is having a mass of 0.1,(100 grams, which is reasonable), the sphere diffuses into the fingers on grasping as in the following picture.
![100g](https://user-images.githubusercontent.com/915936/82249526-8f1dd680-9967-11ea-9f88-9a0a4a791b0b.png)

If I reduce the mass of the finger in URDF to 0.0025 ( 2.5 grams, which is unreasonable), it works fine.
![2_5g](https://user-images.githubusercontent.com/915936/82249611-baa0c100-9967-11ea-97c7-120110d22368.png)

What is the reason for this?
Is there any other fix to this apart from modifying the mass of the finger to a non-realistic value?

**Comment by erwincoumans** [comment]
It is likely a mass ratio issue with the pgs constraint solver. Did you try a smaller time step and more solver iterations (and small residual threshold)? Or direct solver?
We can only give feedback with small self-contained pybullet examples attached as zip. 

If you cannot share, could you modify panda grasp?

https://github.com/bulletphysics/bullet3/tree/master/examples/pybullet/gym/pybullet_robots/panda
